test/confit-test.js

```diff
@@ -470,28 +470,37 @@ test('confit', function (t) {
 
     t.test('precedence', function (t) {
         var factory;
-        var argv = process.argv;
-        var env = process.env;
-
-        process.argv = [ 'node', __filename, '--override=argv'];
-        process.env = {
-            NODE_ENV: 'development',
-            override: 'env',
-            misc: 'env'
-        };
+        var argv;
+        var env;
 
-        factory = confit(path.join(__dirname, 'fixtures', 'defaults'));
-        factory.create(function (err, config) {
-            t.error(err);
-            t.ok(config);
-            t.equal(config.get('override'), 'argv');
-            t.equal(config.get('misc'), 'env');
+        t.test('before', function (t) {
+            argv = process.argv;
+            env = process.env;
             t.end();
         });
 
-        t.on('end', function () {
+        t.test('test', function (t) {
+            process.argv = [ 'node', __filename, '--override=argv'];
+            process.env = {
+                NODE_ENV: 'development',
+                override: 'env',
+                misc: 'env'
+            };
+
+            factory = confit(path.join(__dirname, 'fixtures', 'defaults'));
+            factory.create(function (err, config) {
+                t.error(err);
+                t.ok(config);
+                t.equal(config.get('override'), 'argv');
+                t.equal(config.get('misc'), 'env');
+                t.end();
+            });
+        });
+
+        t.test('after', function (t) {
```

Comment:
It seems strange to use a test for this. Why not just put it inline before t.end up above?